Calibration/IsolatedParticles/src/eHCALMatrix.cc

```diff
@@ -6,6 +6,140 @@
+    for (unsigned int i=0; i<vdets.size(); i++) {
+      std::vector<std::vector<PCaloHit>::const_iterator> hit = spr::findHit(hits, vdets[i]);
+       hitlist.insert(hitlist.end(), hit.begin(), hit.end());
+    }
+
+    double energySum(0);
+    for (unsigned int ihit=0; ihit<hitlist.size(); ihit++) 
+      energySum += hitlist[ihit]->energy();
+
+    // Get hotCell ID
+    dets.clear();
+    std::vector<double> energies;
+    for (unsigned int ihit=0; ihit<hitlist.size(); ihit++) {
+      double energy = hitlist[ihit]->energy();
+      HcalDetId id0 = HcalDetId(hitlist[ihit]->id());
+      if ((id0.subdet() != HcalOuter) || includeHO) {
+      	HcalDetId id1(id0.subdet(),id0.ieta(),id0.iphi(),1);
+        bool found(false);
+        for (unsigned int idet=0; idet<dets.size(); ++idet) {
+          if (id1 == HcalDetId(dets[idet])) {
+	    energies[idet] += energy; 
+	    found = true;
+	    break;
+          }
+        }
+        if (!found) {
+          dets.push_back(DetId(id1));
+          energies.push_back(energy);
+        }
+      }
+    }
+    double energyMax(-99.);
+    for (unsigned int ihit=0; ihit<dets.size(); ihit++) {
+      if (energies[ihit] > energyMax) {
+        energyMax = energies[ihit];
+	hotCell   = HcalDetId(dets[ihit]);
+      }
+    }
+    return energySum;
+  }
+
+  void energyHCALCell(HcalDetId detId, std::vector<PCaloHit>& hits, 
+		      std::vector<std::pair<double,int> >& energyCell, 
+		      int maxDepth, double hbThr, double heThr, double hfThr, 
+		      double hoThr, double tMin, double tMax, bool debug) {
+
+    energyCell.clear();
+    int    subdet  = detId.subdet();	
+    double eThr    = spr::eHCALThreshold(subdet, hbThr, heThr, hfThr, hoThr);
+    bool   hbhe    = (detId.ietaAbs() == 16);
+    int    depthHE = (maxDepth <= 6) ? 3 : 4;
+    if (debug)
+      std::cout << "energyHCALCell: input ID " << detId << " MaxDepth " << maxDepth << " Threshold (E) " << eThr << " (T) " << tMin << ":" << tMax << std::endl;
+    for (int i=0; i<maxDepth; i++) {
+      HcalSubdetector subdet0 = (hbhe) ? ((i+1 >= depthHE) ? HcalEndcap : HcalBarrel) : detId.subdet();
+      HcalDetId hcid(subdet0,detId.ieta(),detId.iphi(),i+1);
+      DetId det(hcid.rawId());
+      std::vector<std::vector<PCaloHit>::const_iterator> hit = spr::findHit(hits, det);
+      double energy(0);
+      for (unsigned int ihit=0; ihit<hit.size(); ++ihit) {
+        if (hit[ihit]->time() > tMin && hit[ihit]->time() < tMax) 
+          energy += hit[ihit]->energy();
+        if (debug)
+          std::cout << "energyHCALCell:: Hit[" << ihit << "] " << hcid << " E " << hit[ihit]->energy() << " t " << hit[ihit]->time() << std::endl;
+      }
+      if (debug)
+	std::cout << "energyHCALCell:: Cell " << hcid << " E " << energy << " from " << hit.size() << " threshold " << eThr << std::endl;
+      if (energy>eThr && hit.size() > 0) {
+        energyCell.push_back(std::pair<double,int>(energy,i+1));
+      }
+    }
+    if (debug) {
```

Comment:
HI @bsunanda - here and elsewhere, please replace the if (debug)s by a compile time switch ( as you usually have)